src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetBuilding.java

```diff
@@ -220,14 +229,11 @@ private boolean build(EntityPlayer player, ItemStack stack) {
         return true;
     }
 
```

Comment:
Suggestion for override implementation...
```suggestion
public static GadgetBuilding getGadget(EntityPlayer p) {
  ItemStack heldItem = player.getHeldItemMainhand();
  if (!(heldItem.getItem() instanceof GadgetBuilding)) {
     heldItem = player.getHeldItemOffhand();
     if (!(heldItem.getItem() instanceof GadgetBuilding)) {
        return null;
     }
  }
  return heldItem;
}
```

Reply:
see https://github.com/Direwolf20-MC/BuildingGadgets/pull/139/commits/85b80b12a0210d16305a0750afc6b2c2fd873a6a